components/search_bar/search_bar.jsx

```diff
@@ -81,15 +81,16 @@ export default class SearchBar extends React.PureComponent {
         }
     }
 
-    componentWillReceiveProps(nextProps) {
-        const {searchTerms} = nextProps;
+    static getDerivedState(props, state) {
+        const {searchTerms} = props;
 
-        this.setState({
-            visibleSearchHintOptions: this.determineVisibleSearchHintOptions(searchTerms),
-        });
+        return {
+            ...state,
```

Comment:
You actually don't need to return unchanged state from `getDerivedStateFromProps`, so this can be removed. If nothing is changing in the state, you just return null.

Reply:
I just pushed a modify that track last changed prop in order to recompute `visibleSearchHintOptions` only when necessary.